streams/jvm/src/main/scala/zio/stream/experimental/Deflate.scala

```diff
@@ -0,0 +1,66 @@
+package zio.stream.experimental
+
+import zio.stream.compression.{CompressionLevel, CompressionStrategy, FlushMode}
+import zio.{Chunk, ZIO, ZManaged}
+
+import java.util.zip.Deflater
+import java.{util => ju}
+import scala.annotation.tailrec
+
+object Deflate {
+  def makeDeflater[Err, Done](
+    bufferSize: Int = 64 * 1024,
+    noWrap: Boolean = false,
+    level: CompressionLevel = CompressionLevel.DefaultCompression,
+    strategy: CompressionStrategy = CompressionStrategy.DefaultStrategy,
+    flushMode: FlushMode = FlushMode.NoFlush
+  ): ZChannel[Any, Err, Chunk[Byte], Done, Err, Chunk[Byte], Done] =
+    ZChannel.managed {
+      ZManaged
+        .make(ZIO.effectTotal {
+          val deflater = new Deflater(level.jValue, noWrap)
+          deflater.setStrategy(strategy.jValue)
+          (deflater, new Array[Byte](bufferSize))
+        }) { case (deflater, _) =>
+          ZIO.effectTotal(deflater.end())
+        }
+    } {
+      case (deflater, buffer) => {
+
+        def loop(): ZChannel[Any, Err, Chunk[Byte], Done, Err, Chunk[Byte], Done] =
+          ZChannel.readWithCause(
+            chunk =>
+              ZChannel.fromEffect {
```

Comment:
Let's use `ZChannel.effectTotal` instead of `ZChannel.fromEffect(ZIO.effectTotal`. Should be a bit faster. Also below.